<launch>

  # 옵션(단위: m, rad, s)
  <arg name="move_dist" default="0.5"/>
  <arg name="move_vel" default="0.3"/>
  <arg name="move_time" default="2.0"/>
  <arg name="turn_vel" default="1.82"/>
  <arg name="wait_1" default="1.0"/>
  <arg name="wait_2" default="3.0"/>
  
  <arg name="lin_vel_joy" default="0.69"/>            # 조이스틱 최고 선속도 0.69
  <arg name="ang_vel_joy" default="1.50"/>            # 조이스틱 최고 각속도 3.67
  
  # 이동로봇 제어
  <node pkg="shared_control" type="simple.py" name="simple" output="screen">
    <param name="move_dist" value="$(arg move_dist)"/>
    <param name="move_vel" value="$(arg move_vel)"/>
    <param name="move_time" value="$(arg move_time)"/>
    <param name="turn_vel" value="$(arg turn_vel)"/>
    <param name="lin_vel_joy" value="$(arg lin_vel_joy)"/>
    <param name="ang_vel_joy" value="$(arg ang_vel_joy)"/>
    <param name="camera" value="camera/color/image"/>
  </node>

  # 인터페이스
  <node pkg="joy" type="joy_node" name="joy_node"/>
  <node pkg="rviz" type="rviz" name="rviz" respawn="true" args="-d $(find shared_control)/rviz/simple.rviz"/>

</launch>
